### DISCUSSION TITLE
Multiple pod template files usage

### DISCUSSION THREAD

**Discussion by burakovsky** [apache/airflow#20149] [Q&A]
I deployed Airflow in Kubernetes using the official Helm chart and want to use different pod templates for some specific DAGs. According to [documentation](https://airflow.apache.org/docs/apache-airflow/stable/executor/kubernetes.html#pod-override) I have 2 options: 
1. use `pod_override` in executor_config section
2. use `pod_template_file` in executor_config section (combine with `pod_override` if necessary)

My question is about the second option: what is the best way to add additional pod templates to scheduler? I found 2 ways to do it:
1. build a custom docker image, which will contain necessary pod templates. The main disadvantage here is that it's not a dynamic template and if I want to check some changes, for example, I have to build a new image and can't just change variables in chart's values.yaml.
2. Add new pod template as `extraConfigMaps` and then mount it as an extra volume for scheduler using `scheduler.extraVolumes` and `scheduler.extraVolumeMounts`. extraConfigMaps can be templated, so it's exactly what I need, but the issue here is that `scheduler.extraVolumes` parameter is not templated. It means, that I can't use {{.Release.Name }} in ConfigMap name which can be a problem when I need to deploy multiple Airflow instances. One more issue: extraConfigmaps have hardcoded helm annotations, which makes it impossible to use if Airflow is deployed using ArgoCD (see #17447 for details)

I didn't find any good explanation of how to manage multiple pod template files in official documentation, probably there are some better solutions I don't know. Will appreciate for any advise!

**Comment by Leoduv** [comment]
Hi, I am asking myself exactly the same questions! [This article](https://medium.com/qonto-way/scaling-airflow-on-kubernetes-lessons-learned-a0d3d0417fc1#:~:text=Pod%20template%20files%20to%20the%20rescue) lead me to believe that there is indeed a way of having multiple pod templates, but I can't figure out what is the best way of doing it.

Even with only one template, the documentation doesn't really show how you are supposed to pass your template to the pods (volume ? in the docker image ? something else ?)

Hope someone can answer

**Reply by potiuk** [reply]
The documentation is quite clear, I tihnk https://airflow.apache.org/docs/apache-airflow/stable/executor/kubernetes.html#pod-template-file 

You have even a very detailed example of pod_override: https://airflow.apache.org/docs/apache-airflow/stable/executor/kubernetes.html#pod-override and below example of using different template for different task

What exactly do you miss in the documentation? Did you try to follow the examples which are explicitly mentioned in the docs?
